frontend/mobilityai/app/src/main/java/com/ai/mobility/mobilityai/MainActivity.java

```diff
@@ -0,0 +1,324 @@
+import android.os.IBinder;
+import android.os.ParcelUuid;
+import android.support.design.widget.FloatingActionButton;
+import android.support.design.widget.Snackbar;
+import android.support.v4.app.FragmentManager;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.DividerItemDecoration;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.support.v7.widget.Toolbar;
+import android.util.Log;
+import android.view.View;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.widget.Button;
+import android.widget.ImageButton;
+
+import com.ai.mobility.mobilityai.MetaMotionDeviceAdapter.OnItemClickListener;
+import com.mbientlab.bletoolbox.scanner.BleScannerFragment;
+import com.mbientlab.metawear.Data;
+import com.mbientlab.metawear.MetaWearBoard;
+import com.mbientlab.metawear.Route;
+import com.mbientlab.metawear.Subscriber;
+import com.mbientlab.metawear.android.BtleService;
+import com.mbientlab.metawear.builder.RouteBuilder;
+import com.mbientlab.metawear.builder.RouteComponent;
+import com.mbientlab.metawear.data.Acceleration;
+import com.mbientlab.metawear.data.AngularVelocity;
+import com.mbientlab.metawear.module.Accelerometer;
+import com.mbientlab.metawear.module.GyroBmi160;
+import com.mbientlab.metawear.module.Led;
+import com.mbientlab.metawear.module.Logging;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Random;
+import java.util.UUID;
+
+import bolts.Continuation;
+import bolts.Task;
+
+public class MainActivity extends AppCompatActivity implements ServiceConnection {//, BleScannerFragment.ScannerCommunicationBus {
+    //Metawear classes
+    private BtleService.LocalBinder serviceBinder;
+    private Led led;
+    private MetaWearBoard board;
+    private Accelerometer accelerometer;
+    private GyroBmi160 gyroscope;
+    private Logging logging;
+
+    private MetaMotionService metaMotion;
+
+    private Button led_on, led_off;
+
+    private static final String TAG = "MobilityAI";
+    private static final int REQUEST_ENABLE_BT = 1, PERMISSION_REQUEST_COARSE_LOCATION= 2;
+
+    //RecylerView
+    private RecyclerView m_bleList;
+    private MetaMotionDeviceAdapter m_adapter;
+    private ArrayList<MetaMotionDevice> m_deviceList;
+
+    private ImageButton m_refreshButton;
+
+
+    //Bluetooth Scanning
+    private BluetoothAdapter m_bluetoothAdapter;
+    private BluetoothLeScanner m_bluetoothLeScanner;
+    private boolean m_isScanning = false;
+    private Handler m_handler;
+    private ScanCallback m_leScanCallback;
+    private HashSet<ParcelUuid> m_filterServiceUuids;
+    private BleScannerFragment.ScannerCommunicationBus commBus= null;
+    private boolean m_isScanReady;
+    private ArrayList<BluetoothDevice> m_devices;
+    private static final long SCAN_PERIOD = 10000L;
+
+
+    //TODO: Remove
+    private Random r = new Random();
```

Comment:
Still need it?

Reply:
Will be removed once the server side part works and I can use that instead of getting random data to populate the pages.